mlos_core/mlos_core/optimizers/bayesian_optimizers/smac_optimizer.py

```diff
@@ -0,0 +1,253 @@
+        seed: Optional[int] = 0,
+        run_name: Optional[str] = None,
+        output_directory: Optional[str] = None,
+        max_trials: int = 100,
+        n_random_init: Optional[int] = 10,
+        n_random_probability: Optional[float] = 0.1,
+    ):
+        super().__init__(parameter_space, space_adapter)
+
+        # pylint: disable=import-outside-toplevel
+        from smac import HyperparameterOptimizationFacade as Optimizer_Smac
+        from smac import Scenario
+        from smac.intensifier.abstract_intensifier import AbstractIntensifier
+        from smac.initial_design import LatinHypercubeInitialDesign
+        from smac.main.config_selector import ConfigSelector
+        from smac.random_design.probability_design import ProbabilityRandomDesign
+        from smac.runhistory import TrialInfo
+
+        # Store for TrialInfo instances returned by .ask()
+        self.trial_info_map: Mapping[ConfigSpace.Configuration, TrialInfo] = {}
+
+        # The default when not specified is to use a known seed (0) to keep results reproducible.
+        # However, if a `None` seed is explicitly provided, we let a random seed be produced by SMAC.
+        # https://automl.github.io/SMAC3/main/api/smac.scenario.html#smac.scenario.Scenario
+        seed = -1 if seed is None else seed
+
+        # Create temporary directory for SMAC output (if none provided)
+        self.temp_output_directory: Optional[TemporaryDirectory] = None
+        if output_directory is None:
+            # pylint: disable=consider-using-with
+            try:
+                self.temp_output_directory = TemporaryDirectory(ignore_cleanup_errors=True)  # Argument added in Python 3.10
+            except TypeError:
+                self.temp_output_directory = TemporaryDirectory()
+            output_directory = self.temp_output_directory.name
+
+        # Instantiate Scenario
+        scenario: Scenario = Scenario(
+            self.optimizer_parameter_space,
+            name=run_name,
+            output_directory=Path(output_directory),
+            deterministic=True,
+            n_trials=max_trials,
+            seed=seed or -1,  # if -1, SMAC will generate a random seed internally
+            n_workers=1,  # Use a single thread for evaluating trials
+        )
+        intensifier: AbstractIntensifier = Optimizer_Smac.get_intensifier(scenario, max_config_calls=1)
+        config_selector: ConfigSelector = ConfigSelector(scenario, retrain_after=1)
+
+        # Customize SMAC's randomized behavior
+        initial_design: Optional[LatinHypercubeInitialDesign] = None
+        if n_random_init is not None:
+            initial_design = LatinHypercubeInitialDesign(scenario=scenario, n_configs=n_random_init)
+        random_design: Optional[ProbabilityRandomDesign] = None
+        if n_random_probability is not None:
+            random_design = ProbabilityRandomDesign(probability=n_random_probability)
+
+        # Create SMAC optimizer
```

Comment:
I would rather remove this and other similar comments that just state the obvious